util/dvsim/SimCfg.py

```diff
@@ -0,0 +1,412 @@
+        self.ts_format = '+%a.%m.%d.%y__%I.%M.%S%p'
+
+        curr_ts = datetime.datetime.now()
+        self.timestamp_long = curr_ts.strftime(self.ts_format_long)
+        self.timestamp = curr_ts.strftime(self.ts_format)
+
+        # Parse the cfg_file file tree
+        self.parse_sim_cfg(args.cfg)
+
+        # If build_unique is set, then add current timestamp to uniquify it
+        if self.build_unique:
+            self.build_dir += "_" + self.timestamp
+
+        # Make substitutions, while ignoring the following wildcards
+        # TODO: Find a way to set these in sim cfg instead
+        ignored_wildcards = ["build_mode",
+                             "index",
+                             "test",
+                             "seed",
+                             "uvm_test",
+                             "uvm_test_seq"]
+        self.__dict__ = find_and_substitute_wildcards(self.__dict__, self.__dict__,
+                                                      ignored_wildcards)
+
+        # Check if there are items to run
+        if self.items == []:
+            log.error("No items provided for running this simulation / regression")
+            sys.exit(1)
+
+        # Use the default build mode for tests that do not specify it
+        if not hasattr(self, "build_mode"):
+            setattr(self, "build_mode", "default")
+
+        self.process_exports()
+
+        # Create objects from raw dicts - build_modes, sim_modes, run_modes,
+        # tests and regressions
+        self.create_objects()
+
+        # Once all objects are constructed, check if the --list option is passed
+        # Is yes, then simply print the items requested and exit.
+        self.print_list()
+
+        # Look at list of items and build the list of tests to run
+        self.deploy = []
+        self.build_list = []
+        self.run_list = []
+        self.create_build_and_run_list()
+
+        # Process reseed override
+        for test in self.run_list:
+            if self.reseed_ovrd != -1:
+                test.reseed = self.reseed_ovrd
+                test.reseed *= self.reseed_multiplier
+
+        # Create deploy objects
+        self.create_deploy_objects()
+
+        # Deploy the builds and runs
+        Deploy.deploy(self.deploy)
+
+
+    def process_exports(self):
+        # Convert 'exports' to dict
+        exports_dict = {}
+        if self.exports != []:
+            for item in self.exports:
+                if type(item) is dict:
+                    exports_dict.update(item)
+                elif type(item) is str:
+                    [key, value] = item.split(':', 1)
+                    if type(key) is not str: key = str(key)
+                    if type(value) is not str: value = str(value)
+                    exports_dict.update({key.strip():value.strip()})
+                else:
+                  log.error("Type error in \"exports\": %s", str(item))
+                  sys.exit(1)
```

Comment:
alignment
run ./util/lintpy.py to fix the format?

Reply:
Done, thanks.